api/src/main/java/org/openmrs/module/dhisconnector/api/impl/DHISConnectorServiceImpl.java

```diff
@@ -1467,6 +1467,20 @@ public String transformToDHISPeriod(Calendar startDate, Calendar endDate, String
 			endDate.set(Calendar.DAY_OF_MONTH, 30);
 			if (lastRun == null || !sdf.format(lastRun).equals(sdf.format(endDate.getTime())))
 				period = startDate.get(Calendar.YEAR) + "Oct";
+		} else if (ReportingPeriodType.FinancialNov.name().equals(periodType)) {
+			sdf = new SimpleDateFormat("yyyy");
+			// Set the start date to year of last Financial November period
+			startDate.add(Calendar.YEAR, startDate.get(Calendar.MONTH) < Calendar.NOVEMBER ? -2 : -1);
```

Comment:
thanks @Piumal1999  do we need this line?

Reply:
Yeah @HerbertYiga . Because the start date depends on the month of the year.
Example:
If the date is 2021/09/05 (a date before November), then the last FinancialNovember period is 2019Nov01-2020Oct31
If the date is 2021/12/05 (a date after November), then the last FinancialNovember period is 2020Nov01-2021Oct31